src/plugins/guided_onboarding/public/components/guide_panel.test.tsx

```diff
@@ -90,143 +59,188 @@ describe('Guided setup', () => {
+      test('shows no manual completion popover if a step is in progress', async () => {
+        const { exists } = await setupComponentWithPluginStateMock(httpClient, {
+          status: 'in_progress',
+          isActivePeriod: true,
+          activeGuide: testGuideStep1InProgressState,
+        });
 
-      expect(find('guideButton').text()).toEqual('Setup guide: step 1');
+        expect(exists('manualCompletionPopoverPanel')).toBe(false);
+      });
     });
 
-    test('shows the step number in the button label if a step is ready to complete', async () => {
-      const { component, find } = testBed;
+    describe('when no guide is active', () => {
+      describe('when in active period', () => {
+        // mock state is by default { status: 'not_started', isActivePeriod: true }
+        test('shows redirect button when no guide has been started yet', () => {
+          const { exists } = testBed;
+          expect(exists('guideButtonRedirect')).toBe(true);
+          expect(exists('guideButton')).toBe(false);
+        });
 
-      await updateComponentWithState(component, mockReadyToCompleteTestGuideState, true);
+        test('shows redirect button when a user quit the guide', async () => {
+          const { exists } = await setupComponentWithPluginStateMock(httpClient, {
+            status: 'quit',
+            isActivePeriod: true,
+          });
 
-      expect(find('guideButton').text()).toEqual('Setup guide: step 2');
-    });
+          expect(exists('guideButtonRedirect')).toBe(true);
+          expect(exists('guideButton')).toBe(false);
+        });
 
-    test('shows the manual completion popover if a step is ready to complete', async () => {
-      const { component, exists } = testBed;
+        test('shows redirect button when a user skipped on the landing page', async () => {
+          const { exists } = await setupComponentWithPluginStateMock(httpClient, {
+            status: 'skipped',
+            isActivePeriod: true,
+          });
 
-      await updateComponentWithState(component, mockReadyToCompleteTestGuideState, false);
+          expect(exists('guideButtonRedirect')).toBe(true);
+          expect(exists('guideButton')).toBe(false);
+        });
 
-      expect(exists('manualCompletionPopover')).toBe(true);
-    });
+        test('hides the button if the user completed a guide', async () => {
+          const { exists } = await setupComponentWithPluginStateMock(httpClient, {
+            status: 'complete',
+            isActivePeriod: true,
+          });
 
-    test('shows no manual completion popover if a step is in progress', async () => {
-      const { component, exists } = testBed;
+          expect(exists('guideButtonRedirect')).toBe(false);
+          expect(exists('guideButton')).toBe(false);
+        });
+      });
 
-      await updateComponentWithState(component, mockInProgressTestGuideState, false);
+      describe('when not in active period', () => {
```

Comment:
Does it make sense to add a test that the button is still visible if a guide is in progress and not in the active period?